stl/src/mutex.cpp

```diff
@@ -77,26 +77,28 @@ void _Mtx_destroy(_Mtx_t mtx) { // destroy mutex
 }
 
 static _Thrd_result mtx_do_lock(_Mtx_t mtx, const _timespec64* target) { // lock mutex
+    // TRANSITION, ABI: the use of `const _timespec64*` is preserved for `_Mtx_timedlock`
+    const long current_thread_id = static_cast<long>(GetCurrentThreadId());

```

Comment:
Nitpick:
No need to repeat the type twice, `auto` is enough.

Reply:
While that doesn't save any characters, it is indeed consistent with our usual conventions.